pkg/virt-handler/vm.go

```diff
@@ -1904,18 +1904,40 @@ func (d *VirtualMachineController) isPreMigrationTarget(vmi *v1.VirtualMachineIn
 }
 
 func (d *VirtualMachineController) checkNetworkInterfacesForMigration(vmi *v1.VirtualMachineInstance) error {
-	networks := map[string]*v1.Network{}
-	for _, network := range vmi.Spec.Networks {
-		networks[network.Name] = network.DeepCopy()
+	err := validatePodNetworkInterfaceUsesMasqueradeBinding(vmi)
+	if err != nil {
+		return err
 	}
+
+	return nil
+}
+
+func validatePodNetworkInterfaceUsesMasqueradeBinding(vmi *v1.VirtualMachineInstance) error {
+	podNetwork := getVMIPodNetwork(vmi)
+
+	interfacesLookup := map[string]v1.Interface{}
 	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
-		if iface.Masquerade == nil && networks[iface.Name].Pod != nil {
-			return fmt.Errorf("cannot migrate VMI which does not use masquerade to connect to the pod network")
-		}
+		interfacesLookup[iface.Name] = iface
 	}
+
+	if podNetwork != nil && interfacesLookup[podNetwork.Name].Masquerade == nil {
+		return fmt.Errorf("cannot migrate VMI which does not use masquerade to connect to the pod network")
+	}
+
 	return nil
 }
 
+func getVMIPodNetwork(vmi *v1.VirtualMachineInstance) *v1.Network {
```

Comment:
`get` ==> `lookup`

I suggest you return only the name, as this is the only think you need.

Reply:
Done